### ISSUE TITLE
Notify screen reader users of link type

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#1908]
# Feature Request

## Description
We have various link types defined in our design standard the visually communicate what type of link it is. I'm curious if we should extend this functionality to screen reader users. I'm thinking adding visually hidden text to the hyperlink component that indicates the same meaning the icon in the hyperlink variant is used to indicate. Similarly, disabled links aren't a thing in HTML, should we communicate that link is "disabled" to screen reader users or would that just be confusing.

Related discussion: https://ux.stackexchange.com/questions/105295/links-external-links-and-how-screen-readers-perceive-them

**Comment by bjankord** [comment]
cc/ @neilpfeiffer @scottwilmarth

**Comment by neilpfeiffer** [comment]
Need to decide best A11Y enhancement method to indicate hyperlink types (external link, audio, video, photo, document). 

Could either be 
- setting `role="img"` on SVG, and providing ALT content -- source: https://css-tricks.com/accessible-svgs/
- or including `visually-hidden-text` with similar phrase/explanation as the ALT content that indicates type of hyperlink, keeps image as aria-hidden to screenreader -- source: https://codersblock.com/blog/external-links-new-tabs-and-accessibility/

**Comment by neilpfeiffer** [comment]
Missing Design Input: 
(1) Need to identify the needed strings (and translations) to add.
(2) Final method to disclose to screenreader.
